### PR TITLE
[17688] CalendarWidget Refactoring

### PR DESCRIPTION
**Pull request by LanceGundersen**
## Description
Rename onClickNext and onClickPrev to onNextMonth and onPreviousMonth

- Rename onClickNext to onNextMonth
- Rename onClickPrev to onPreviousMonth
- Add JSDoc comments
- Add typedef for FHIR slot
- Changed string to number comparison to string to string comparison

JSDoc Updates: http://localhost:63342/AppointmentsPageV2.jsx/src/applications/vaos/docs/module-components_calendar.html 

## Testing done
local

## Screenshots
n/a

## Acceptance criteria
- [ ] Rename onClickNext and onClickPrev to onNextMonth and onPreviousMonth

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/components/calendar/CalendarWidget.jsx [hunk 1]
```diff
@@ -46,10 +74,16 @@ export function getMaxMonth(maxDate, startMonth) {
...
+ * Gets the initial blank cells
+ *
+ * @param {Moment} momentDate A given moment date
+ * @returns {Array} Array of blanks
+ */
 function getInitialBlankCells(momentDate) {
   const firstWeekday = getWeekdayOfFirstOfMonth(momentDate);
 
-  if (firstWeekday === 0 || firstWeekday === 6) {
+  if (firstWeekday === '0' || firstWeekday === '6') {
```
Comment (LanceGundersen) [inline src/applications/vaos/components/calendar/CalendarWidget.jsx:86]: Changed to strings

File: src/applications/vaos/components/calendar/CalendarWidget.jsx [hunk 2]
```diff
@@ -144,8 +237,8 @@ export default function CalendarWidget({
   maxSelectionsError = "You've exceeded the maximum number of selections",
   minDate,
   onChange,
-  onClickNext,
-  onClickPrev,
+  onClickNext: onNextMonth,
```
Comment (jbalboni) [inline src/applications/vaos/components/calendar/CalendarWidget.jsx:240]: It looks like this just renames the prop when using inside the component. Ideally, we would change these prop names for consumers of this component, too, and update the three pages that use the CalendarWidget.

Comment (LanceGundersen) [reply src/applications/vaos/components/calendar/CalendarWidget.jsx]: Makes sense, fixed.

File: src/applications/vaos/components/calendar/CalendarWidget.jsx (hunk 1 above)
Comment (vbahinwillit) [reply src/applications/vaos/components/calendar/CalendarWidget.jsx:86]: I see why the change to String but would it be better if it was a number? Reason being is that days of the week are typically numbers (0 - 6). Just a thought.

Comment (vbahinwillit) [reply src/applications/vaos/components/calendar/CalendarWidget.jsx:86]: This following would not be needed as well, 

let dayOfWeek = **Number**(getWeekdayOfFirstOfMonth(momentDate));

**Review by vbahinwillit — APPROVED** [review]
LGTM! Loving the documentation!!!!

**Comment by LanceGundersen** [comment]
@vbahinwillit good catch on the number issue there. I fixed the symptom not the problem! 🙈

**Review by jbalboni — APPROVED** [review]

File: src/applications/vaos/components/calendar/CalendarWidget.jsx [hunk 3]
```diff
@@ -34,10 +34,10 @@ function pad(num, size) {
  * Gets the weekday of the first month
  *
  * @param {Moment} momentDate A given moment date
- * @returns {string} A formatted date e.g. 3
+ * @returns {number} A number of the week e.g. 0-6
  */
 function getWeekdayOfFirstOfMonth(momentDate) {
-  return momentDate.startOf('month').format('d');
+  return Number(momentDate.startOf('month').format('d'));
```
Comment (vbahinwillit) [inline src/applications/vaos/components/calendar/CalendarWidget.jsx:40]: momentDate.startOf('month').day();

**Pull request merged**